examples/github-deploy-only.yml

```diff
@@ -0,0 +1,82 @@
+name: Build and upload to PyPI
+
+on: [push, pull_request]
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, windows-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install Python
+      with:
+        python-version: '3.7'
+
+    - name: Install cibuildwheel
+      run: |
+        python -m pip install cibuildwheel==1.4.2
+
+    - name: Install Visual C++ for Python 2.7
+      if: startsWith(matrix.os, 'windows')
+      run: |
+        choco install vcpython27 -f -y
+
+    - name: Build wheels
+      run: |
+        python -m cibuildwheel --output-dir wheelhouse
+
+    - uses: actions/upload-artifact@v1
+      with:
+        name: wheels
+        path: ./wheelhouse
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install Python
+      with:
+        python-version: '3.7'
+
+    - name: Build sdist
+      run: python setup.py sdist -d wheelhouse/
+
+    - uses: actions/upload-artifact@v2
+      with:
+        path: wheelhouse/*.tar.gz
+
+  upload_pypi:
+    needs: [build_wheels, build_sdist]
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.8"
+
+    - name: Install dependencies
+      run: python -m pip install twine
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: artifact
+        path: wheelhouse
+
+    - run: twine check wheelhouse/*
+
+    - uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.pypi_password }}
+        packages_dir: ./wheelhouse/
+        # To test: repository_url: https://test.pypi.org/legacy/
+      if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
@henryiii I assume this is checking for git tags to only upload on a release, but what exactly do the two parts check/mean?

Reply:
I don't think this will work as you have `on: push, pull`
https://github.com/joerick/cibuildwheel/pull/362/files#diff-fe7f06db19278183fd8cb674f1009afcR3 

See https://github.com/htm-community/htm.core/actions/runs/122069673/workflow#L12-L16

Reply:
You need to have `on: release` enabled too. You probably normally want to run this on tags & releases, and not on push and pull_requests. I don't have this fully tested in Scikit-HEP yet; several packages have adopted the pure-python version, but only one is on the cibuildwheel for now and hasn't had a release yet, but the idea was that only when a GitHub Release is made that you push to PyPI.

Reply:
Oh, wait, so I change `on: push, pull` to `on: push, pull, release`?

Apart from that, we can merge it, and if you figure out more, could you let us know?

Reply:
Updated. Like this?

Reply:
Sure. I'll probably be trying to make a release before SciPy, and some of the other packages that have binary components may pick it up.

Reply:
Thanks!

Reply:
Yes, that looks right.